### PR TITLE
Add argument to remove sources content from react native source maps

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What and why?

Enable removal of sources content from react native source maps.
This can reduce the size of the sourcemap by 60%.
However doing so will result in not having the code snippet next to the unminified error, but that is considered nice to have for now.

Solves https://github.com/DataDog/dd-sdk-reactnative/issues/302

### How?

Also added some optional arguments to the xcode command.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] \[_Synthetics_\] New releases of `datadog-ci` MUST be followed by a release of Synthetics CI integrations

### CODE REVIEW

**Review by lefebvree — APPROVED** [review]

File: src/commands/react-native/README.md [hunk 1]
```diff
@@ -42,22 +42,24 @@ To upload the source maps for Android, run this command:
...
+| `--disable-git`            | False   | Prevents the command from invoking git in the current working directory and sending repository-related data to Datadog (such as the hash, remote URL, and paths within the repository of sources referenced in the source map).   |
+| `--dry-run`                | False   | It runs the command without the final step of uploading. All other checks are performed.                                                                                                                                          |
+| `--repository-url`         | Empty   | Overrides the remote repository with a custom URL. For example, `https://github.com/my-company/my-project`.                                                                                                                       |
+| `--remove-sources-content` | False   | Remove the `"sourcesContent"` part of the source map files. Doing so greatly reduces the size of your files while still keeping the unminification, but it also removes the code snippet next to the unminified error in Datadog. |
```
Comment (maycmlee) [inline src/commands/react-native/README.md:61]:
```suggestion
| `--remove-sources-content` | False   | Removes the `"sourcesContent"` part of the source map files. This reduces the size of your files while still keeping the unminification, but it also removes the code snippet next to the unminified error in Datadog. |
```

File: src/commands/react-native/README.md [hunk 2]
```diff
@@ -42,22 +42,24 @@ To upload the source maps for Android, run this command:
...
+| `--config`                 | Empty   | Path to your `datadog-ci.json` file if not at the root of your project                                                                                                                                                            |
```
Comment (maycmlee) [inline src/commands/react-native/README.md:62]:
```suggestion
| `--config`                 | Empty   | The path to your `datadog-ci.json` file, if it is not at the root of your project.                                                                                                                                                            |
```

File: src/commands/react-native/README.md [hunk 3]
```diff
@@ -112,24 +113,25 @@ datadog-ci react-native codepush --platform ios --service com.company.app --bund
...
+| `--repository-url`         | Empty   | Overrides the remote repository with a custom URL. For example, `https://github.com/my-company/my-project`.                                                                                                                                                                                                   |
+| `--build-version`          | 1       | Used to avoid overwriting your source maps by accident. Only one upload is needed for a specific `build-version` and `service` combination. Subsequent uploads are ignored until the `build-version` changes. This should not be necessary for CodePush unless you uploaded the wrong source maps by mistake. |
+| `--remove-sources-content` | False   | Remove the `"sourcesContent"` part of the source map files. Doing so greatly reduces the size of your files while still keeping the unminification, but it also removes the code snippet next to the unminified error in Datadog.                                                                             |
```
Comment (maycmlee) [inline src/commands/react-native/README.md:133]:
```suggestion
| `--remove-sources-content` | False   | Removes the `"sourcesContent"` part of the source map files. This reduces the size of your files while still keeping the unminification, but it also removes the code snippet next to the unminified error in Datadog.                                                                             |
```

File: src/commands/react-native/README.md [hunk 4]
```diff
@@ -112,24 +113,25 @@ datadog-ci react-native codepush --platform ios --service com.company.app --bund
...
+| `--config`                 | Empty   | Path to your `datadog-ci.json` file if not at the root of your project                                                                                                                                                                                                                                        |
```
Comment (maycmlee) [inline src/commands/react-native/README.md:134]:
```suggestion
| `--config`                 | Empty   | The path to your `datadog-ci.json` file, if it is not at the root of your project.                                                                                                                                                                                                                                        |
```

File: src/commands/react-native/README.md [hunk 5]
```diff
@@ -186,17 +187,26 @@ export NODE_BINARY=node
...
 | `--composeSourcemapsPath` | `../node_modules/react-native/scripts/compose-source-maps.js` | If you use Hermes, you need to compose the source maps after the bundle phase. Only use this argument if your node modules are not on the same level as the `ios` folder. |
 
+The following optional parameters are available:
+
+| Parameter                  | Default | Description                                                                                                                                                                                                                       |
+| -------------------------- | ------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `--disable-git`            | False   | Prevents the command from invoking git in the current working directory and sending repository-related data to Datadog (such as the hash, remote URL, and paths within the repository of sources referenced in the source map).   |
```
Comment (maycmlee) [inline src/commands/react-native/README.md:205]:
```suggestion
| `--disable-git`            | False   | Prevents the command from invoking git in the current working directory and from sending repository-related data to Datadog (such as the hash, remote URL, and paths within the repository of sources referenced in the source map).   |
```

File: src/commands/react-native/README.md [hunk 6]
```diff
@@ -186,17 +187,26 @@ export NODE_BINARY=node
...
+| `--repository-url`         | Empty   | Overrides the remote repository with a custom URL. For example, `https://github.com/my-company/my-project`.                                                                                                                       |
+| `--remove-sources-content` | False   | Remove the `"sourcesContent"` part of the source map files. Doing so greatly reduces the size of your files while still keeping the unminification, but it also removes the code snippet next to the unminified error in Datadog. |
```
Comment (maycmlee) [inline src/commands/react-native/README.md:207]:
```suggestion
| `--remove-sources-content` | False   | Removes the `"sourcesContent"` part of the source map files. This reduces the size of your files while still keeping the unminification, but it also removes the code snippet next to the unminified error in Datadog. |
```

File: src/commands/react-native/README.md [hunk 7]
```diff
@@ -186,17 +187,26 @@ export NODE_BINARY=node
...
+| `--config`                 | Empty   | Path to your `datadog-ci.json` file if not at the root of your project                                                                                                                                                            |
```
Comment (maycmlee) [inline src/commands/react-native/README.md:208]:
```suggestion
| `--config`                 | Empty   | The path to your `datadog-ci.json` file, if it is not at the root of your project.                                                                                                                                                            |
```

**Review by maycmlee — COMMENTED** [review]
Left some feedback!

**Review by maycmlee — APPROVED** [review]

**Pull request merged**